# Generated by Django 3.2.3 on 2021-10-12 11:24

import logging

from django.db import migrations
from django.db.utils import IntegrityError

logger = logging.getLogger(__name__)


def set_chain_in_transaction(apps, schema_editor):
    # We can't import the Person model directly as it may be a newer
    # version than this migration expects. We use the historical version.
    Transaction = apps.get_model("blockchain", "Transaction")

    for transaction in Transaction.objects.filter(chain_id=None).select_related("block"):
        transaction.chain_id = transaction.block.chain_id
        try:
            transaction.save()
        except IntegrityError as exc:
            logger.warning(f"{transaction} will not be migrated: {exc}")


class Migration(migrations.Migration):

    dependencies = [
        ("blockchain", "0002_auto_20211012_1123"),
    ]

    operations = [migrations.RunPython(set_chain_in_transaction, migrations.RunPython.noop)]
